### PR TITLE
Issue/74 basic patient data form

### LINKED ISSUE #73: Extend user settings page for patient data form
The patient data should be gathered via a basic form. For now, we put this into the user settings.

### LINKED ISSUE #74: Basic patient data form
For now, every attribute should be optional.

- Name
- Date of Birth
- Organ Donor
  - This should be a bool
- Weight
- Height
- Primary language

### PR DESCRIPTION
**Pull request by Florian-Sabonchi**
Closes #73
Closes #74

### CODE REVIEW

**Comment by Florian-Sabonchi** [comment]
image

File: lib/pages/emergency_pass.dart [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+                        Padding(
+                          padding: const EdgeInsets.symmetric(vertical: paddingSmall),
+                          child: TextField(
+                            readOnly: true,
+                            onTap: () async {
+                              DateTime? selectedDate = await showDatePicker(
+                                context: context,
+                                initialDate: DateTime.now(),
+                                firstDate: DateTime(1900),
+                                lastDate: DateTime(2100),
```
Comment (DasProffi) [inline lib/pages/emergency_pass.dart:54]: lastdate should be datetime now

File: lib/pages/emergency_pass.dart [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                        Padding(padding: const EdgeInsets.symmetric(vertical: paddingSmall),
+                          child: TextField(
+                            decoration: InputDecoration(
+                              border: const OutlineInputBorder(),
+                              prefixIcon: const Icon(Icons.height),
+                              labelText: AppLocalizations.of(context)!.height,
+                              hintText: AppLocalizations.of(context)!.height,
+                            ),
+                          ),
+                        ),
```
Comment (DasProffi) [inline lib/pages/emergency_pass.dart:134]: weight, height should be [rangesliders](https://api.flutter.dev/flutter/material/RangeSlider-class.html) or have only number, float input. The unit of measurement like meter or feet would be nice in a suffix.

**Review by DasProffi — CHANGES_REQUESTED** [review]

File: lib/pages/emergency_pass.dart (hunk 1 above)
Comment (use-to) [reply lib/pages/emergency_pass.dart:54]: This reminds me that we could eventually have patient data from multiple people stored on one device....

Comment (Florian-Sabonchi) [reply lib/pages/emergency_pass.dart:55]: We should discuss again

**Review by DasProffi — APPROVED** [review]

**Comment by MaxSchaefer** [comment]
Nice @Florian-Sabonchi, the form works now correctly. Some questions:
- Can we use a select instead of a text field for the "mother tongue"-Field? The options should be a list of possible languages.
- How can the user unset the values Birthday and Organ donor? The values of the other fields can be deleted. A boolean field is either true or false, but null/not set must also be an option.

File: lib/pages/emergency_pass.dart [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+                              ),
+                              prefixIcon: const Icon(Icons.language),
+                              labelText: AppLocalizations.of(context)!.language,
+                              hintText: AppLocalizations.of(context)!.language,
+                            ),
+                            child: LanguagePickerDropdown(
+                                itemBuilder: _buildDropdownItem,
+                                initialValue: Languages.german,
+                                onValuePicked: (Language language) {
+                                  print(language.name);
```
Comment (use-to) [inline lib/pages/emergency_pass.dart:72]:
```
Don't, don't, don't, don't
Don't you, forget about me
```

**Review by use-to — CHANGES_REQUESTED** [review]

File: lib/pages/emergency_pass.dart (hunk 3 above)
Comment (use-to) [inline lib/pages/emergency_pass.dart:72]:
```suggestion
```

**Review by MaxSchaefer — APPROVED** [review]
Nice! Thank you, this works and looking great. In the future, we should use the "x" at the end of an input field more broadly.

**Review by use-to — APPROVED** [review]

**Pull request merged**